simple/tests/stats/cache_test.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import shutil
+import sqlite3
+import tempfile
+import unittest
+from unittest import mock
+
+from freezegun import freeze_time
+import pandas as pd
+from parameterized import parameterized
+from stats import schema
+from stats import schema_constants as sc
+from stats.cache import _generate_svg_cache_internal
+from stats.data import Observation
+from stats.data import Triple
+from stats.db import create_db
+from stats.db import create_main_dc_config
+from stats.db import create_sqlite_config
+from stats.db import get_cloud_sql_config_from_env
+from stats.db import get_sqlite_config_from_env
+from stats.db import ImportStatus
+from stats.db import to_observation_tuple
+from stats.db import to_triple_tuple
+from tests.stats.test_util import is_write_mode
+
+_TEST_DATA_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                              "test_data", "cache")
+_INPUT_DIR = os.path.join(_TEST_DATA_DIR, "input")
+_EXPECTED_DIR = os.path.join(_TEST_DATA_DIR, "expected")
+
+
+def _compare_files(test: unittest.TestCase, output_path: str,
+                   expected_path: str, test_name: str):
+  with open(output_path) as gotf:
+    got = gotf.read()
+    with open(expected_path) as wantf:
+      want = wantf.read()
+      test.assertEqual(got, want, test_name)
+
+
+def _read_triples_csv(path: str) -> list[Triple]:
+  df = pd.read_csv(path, keep_default_na=False)
+  return [Triple(**kwargs) for kwargs in df.to_dict(orient='records')]
```

Comment:
This is the 3rd copy _read_triples_csv and _compare_files in this repo.  We should consolidate them into common test utils / classes.

Reply:
Good call. There are actually more than 3.

I've put them in a test util and am using them in this test. Will do a follow up PR to make all other tests use them.